pilot/pkg/status/manager.go

```diff
@@ -31,21 +34,38 @@ import (
 // when an individual resource is ready to be updated with the relevant data.
 type Manager struct {
 	// TODO: is Resource the right abstraction?
-	store   model.ConfigStore
-	workers WorkerQueue
+	store        model.ConfigStore
+	workers      WorkerQueue
+	lastStatuses sync.Map
 }
 
 func NewManager(store model.ConfigStore) *Manager {
+	manager := &Manager{
+		store:        store,
+		lastStatuses: sync.Map{},
+	}
 	writeFunc := func(m *config.Config, istatus any) {
-		scope.Debugf("writing status for resource %s/%s", m.Namespace, m.Name)
 		status := istatus.(GenerationProvider)
 		m.Status = status.Unwrap()
+		lastStatus, ok := manager.lastStatuses.Load(m.Key())
+		if !ok {
+			cfg := store.Get(m.GroupVersionKind, m.Name, m.Namespace)
+			lastStatus = cfg.Status
+			manager.lastStatuses.Store(m.Key(), lastStatus)
+		}
+		if reflect.DeepEqual(lastStatus, m.Status) {
```

Comment:
usually m.status is going to be a protobuf, doing reflect.DeepEqual on protobuf is not safe (will usually return false even when it shouldn't)